src/modules/job-info/job_state.c

```diff
@@ -174,6 +224,39 @@ struct job_state_ctx *job_state_create (flux_t *h)
         goto error;
     }
 
+    /* sets jsctx->events */
+    if (setup_events_listener (jsctx) < 0)
+        goto error;
+
+    jsctx->events_reconnect_waittime = EVENTS_RECONNECT_WAITTIME;
+
+    if (flux_conf_unpack (flux_get_conf (h),
+                          &err,
+                          "{s?{s?s}}",
+                          "job-info",
+                            "events_reconnect_waittime",
+                            &events_reconnect_waittime_str) < 0) {
+        flux_log (h, LOG_ERR,
+                  "error reading job-info config: %s",
+                  err.errbuf);
+    }
+
+    if (events_reconnect_waittime_str) {
+        if (fsd_parse_duration (events_reconnect_waittime_str,
+                                &jsctx->events_reconnect_waittime) < 0)
+            flux_log_error (h, "error parsing events_reconnect_waittime");
+    }
+
+    if (!(w = flux_timer_watcher_create (r,
+                                         jsctx->events_reconnect_waittime,
+                                         0.,
+                                         setup_events_timer_cb,
+                                         jsctx))) {
+        flux_log_error (h, "flux_timer_watcher_create");
+        goto error;
+    }
```

Comment:
I think you should just call `flux_reactor_stop_error()` if you ever get an error on the job manager event stream instead of trying to recover from it.  It seems unreasonable to expect modules to recover if services they use are unloaded.  If we decided to add this kind of recovery code across the board, we would expend a lot of effort that would be better applied toward formalizing the "used by" relationship between modules and preventing an "in use" module from being unloaded.

Reply:
In `job_events-continuation()`, only the first `flux_rpc_get_unpack()` handles an error by stopping the reactor, but every other error in that function would also be a malformed message, so shouldn't they all behave the same?

Also there is logging for each unique "protocol error" failure there.  I don't think that's useful for all these unlikely errors.

`job_events_continuation()` calls `job_events_priority()` which does some more parsing, but that function returns void.  Suggest you make that function return -1 on failure, and similarly, the logging of each detailed protocol failure probably isn't helpful.

Reply:
ahhh correct.  I was "following" what I did before when checking for ENOSYS, but you are correct, the error should be handled everywhere.